executor/executor.go

```diff
@@ -926,7 +931,22 @@ func (e *SelectLockExec) Next(ctx context.Context, req *chunk.Chunk) error {
 		lockWaitTime = kv.LockNoWait
 	}
 
-	return doLockKeys(ctx, e.ctx, newLockCtx(e.ctx.GetSessionVars(), lockWaitTime), e.keys...)
+	err = doLockKeys(ctx, e.ctx, newLockCtx(e.ctx.GetSessionVars(), lockWaitTime), e.keys...)
+	if err == nil && len(e.keys) > 0 {
```

Comment:
Since `tblID2Handle` will not change during execution, it's better to do the update one time instead of every `Next` call.